src/Microsoft.NetCore.Analyzers/Core/Security/UseXmlReaderForDataSetReadXml.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.NetCore.Analyzers.Security.Helpers;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class UseXmlReaderForDataSetReadXml : UseXmlReaderBase
+    {
+        internal static readonly DiagnosticDescriptor RealRule =
+            SecurityHelpers.CreateDiagnosticDescriptor(
+                "CA5366",
+                nameof(SystemSecurityCryptographyResources.UseXmlReaderForDataSetReadXml),
+                nameof(SystemSecurityCryptographyResources.UseXmlReaderMessage),
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                helpLinkUri: null,
+                nameof(SystemSecurityCryptographyResources.UseXmlReaderDescription));
+
```

Comment:
Did you mean to put this description be on all the other rules as well?

Reply:
Yes, all of them is recommended to use XmlReader, cause only XmlReader can deal with entrusted data.

Reply:
I didn't see it in the other rules.